test/box-luatest/error_subsystem_improvements_test.lua

```diff
@@ -169,3 +171,80 @@ g.test_unpack = function()
     t.assert_equals(u.custom_type, nil)
     t.assert_equals(type(u.code), 'number')
 end
+
+local prevprev = box.error.new{reason = "prevprev"}
+local prev = box.error.new{reason = "prev", prev = prevprev}
+local cur = box.error.new{reason = "cur", prev = prev}
+
+-- Test the increased `box.error` serialization verbosity (gh-9105).
+g.test_increased_error_serialization_verbosity = function()
```

Comment:
(Side thought.)

For my taste, the test cases are a bit too algebraic. A kind of 'assuming that :unpack() works correctly, we verify that serialize/tostring is a transformation of its output to the following shape'. I would just pick up one/two sample errors and a few fields from them and check that __serialize/__tostring gives me a correct result.

This approach doesn't decrease the power of the test case, but verifies more. It is also more human friendly. And, in this case, it wouldn't depend on stability of the order of fields serialization into json/yaml.

Anyway, the present test cases do the job and I don't think that being picky about tests is productive. Let's take this comment as a side note.